src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -697,7 +712,8 @@ private static ImmutableArray<RefKind> GetRefKindsOrNull(ArrayBuilder<RefKind> r
                 // Except for async stack spilling which needs to know whether arguments were originally passed as "In" and must obey "no copying" rule.
                 if (paramRefKind == RefKind.In)
                 {
-                    argRefKind = argRefKind == RefKind.None ? paramRefKind : RefKindExtensions.StrictIn;
+                    Debug.Assert(argRefKind == RefKind.None || argRefKind == RefKind.In);
```

Comment:
Should we also assert here that `argRefKind != RefKind.None || !(argumentsBinderOpt is null)`. Once there is any implicit `in` then we need to ensure that a binder was passed down.